org.eclipse.xtext/src/org/eclipse/xtext/resource/impl/EObjectDescriptionLookUp.java

```diff
@@ -115,7 +115,8 @@ protected Map<QualifiedName, List<IEObjectDescription>> getNameToObjects() {
 		if (nameToObjects == null) {
 			synchronized (this) {
 				if (nameToObjects == null) {
-					Map<QualifiedName, List<IEObjectDescription>> nameToObjects = Maps.newHashMapWithExpectedSize(allDescriptions.size());
+					Map<QualifiedName, List<IEObjectDescription>> nameToObjects = new HashMap<>(
```

Comment:
I don't like repeating that arithmetic expression all over the place. A logical step would be to extract a factory method. Let's call it `newHashMapWithExpectedSize` and put it into a type `Maps`. Oh wait.

Sorry for being sarcastic here, but I'm a little tired of the perceived panic mode when it comes to Guava dependencies.

Reply:
> Sorry for being sarcastic here, but I'm a little tired of the perceived panic mode when it comes to Guava dependencies.

It's not panic, it's more annoyance if you can't upgrade because of conflicting version in unfortunate situations. And I'm tired of it as well, which is why I try to work on a solution.